libraries/shared/watcher/storage_watcher.go

```diff
@@ -47,155 +45,110 @@ func (e ErrHeaderMismatch) Error() string {
-	Queue                     storage.Queue
 	HeaderRepository          datastore.HeaderRepository
-	StorageDiffRepository     storage.DiffRepository
 	KeccakAddressTransformers map[common.Hash]transformer.StorageTransformer // keccak hash of an address => transformer
+	RetryInterval             time.Duration
+	StorageDiffRepository     storage.DiffRepository
 }
 
-func NewStorageWatcher(fetcher fetcher.IStorageFetcher, db *postgres.DB) StorageWatcher {
-	queue := storage.NewStorageQueue(db)
+func NewStorageWatcher(db *postgres.DB, retryInterval time.Duration) StorageWatcher {
 	headerRepository := repositories.NewHeaderRepository(db)
 	storageDiffRepository := storage.NewDiffRepository(db)
 	transformers := make(map[common.Hash]transformer.StorageTransformer)
 	return StorageWatcher{
 		db:                        db,
-		StorageFetcher:            fetcher,
-		Queue:                     queue,
 		HeaderRepository:          headerRepository,
-		StorageDiffRepository:     storageDiffRepository,
 		KeccakAddressTransformers: transformers,
+		RetryInterval:             retryInterval,
+		StorageDiffRepository:     storageDiffRepository,
 	}
 }
 
-func (storageWatcher StorageWatcher) AddTransformers(initializers []transformer.StorageTransformerInitializer) {
+func (watcher StorageWatcher) AddTransformers(initializers []transformer.StorageTransformerInitializer) {
 	for _, initializer := range initializers {
-		storageTransformer := initializer(storageWatcher.db)
-		storageWatcher.KeccakAddressTransformers[storageTransformer.KeccakContractAddress()] = storageTransformer
+		storageTransformer := initializer(watcher.db)
+		watcher.KeccakAddressTransformers[storageTransformer.KeccakContractAddress()] = storageTransformer
+	}
+}
+
+func (watcher StorageWatcher) Execute() error {
+	for {
+		err := watcher.transformDiffs()
+		if err != nil {
+			logrus.Errorf("error transforming diffs: %s", err.Error())
+			return err
+		}
 	}
 }
 
-func (storageWatcher StorageWatcher) Execute(queueRecheckInterval time.Duration) error {
-	ticker := time.NewTicker(queueRecheckInterval)
-	diffsChan := make(chan types.RawDiff)
-	errsChan := make(chan error)
+func (watcher StorageWatcher) transformDiffs() error {
+	diffs := make(chan types.PersistedDiff)
+	errs := make(chan error)
+	done := make(chan bool)
 
-	defer close(diffsChan)
-	defer close(errsChan)
+	defer close(diffs)
+	defer close(errs)
+	defer close(done)
 
-	go storageWatcher.StorageFetcher.FetchStorageDiffs(diffsChan, errsChan)
+	go watcher.StorageDiffRepository.GetNewDiffs(diffs, errs, done)
 
 	for {
 		select {
-		case fetchErr := <-errsChan:
-			logrus.Warnf("error fetching storage diffs: %s", fetchErr.Error())
-			return fetchErr
-		case diff := <-diffsChan:
-			storageWatcher.processRow(diff)
-		case <-ticker.C:
-			storageWatcher.processQueue()
+		case diff := <-diffs:
+			err := watcher.transformDiff(diff)
+			if err != nil {
+				logrus.Infof("error transforming diff: %s", err.Error())
```

Comment:
Opted to just log any errors in the process of transforming a diff, since a lot of the errors are quite common - e.g. no matching header (yet), no matching known storage key (yet), etc. The one place where we might want something more complex is if we fail to mark a diff checked after successfully transforming it, since I think that means the diff will get stuck with `checked` set to false and subsequent attempts to transform always failing on a uniqueness constraint violation. But I figure maybe we'll end up addressing that scenario whenever we setup a process to prune obsolete diffs?